### PR TITLE
test: add smoke-test to CI

### LINKED ISSUE #382: Add a smoke test that runs the application in the CI

### PR DESCRIPTION
**Pull request by MegaRedHand**
Fixes #382

We aren't currently starting up the application in the CI. This PR fixes that.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -102,12 +102,52 @@ jobs:
...
+        protoc --elixir_out=./lib proto/libp2p.proto
+        mix deps.get
+    - name: Set up cargo cache
+      uses: Swatinem/rust-cache@v2
+      with:
+        workspaces: ${{ env.RUST_WORKSPACES }}
+    - name: Compile Elixir (Warnings as errors)
+      run: mix compile --warnings-as-errors
+    - name: Run the node
+      run: mix run -- --checkpoint-sync https://sync-mainnet.beaconcha.in/
```
Comment (mpaulucci) [inline .github/workflows/ci.yml:143]: who stops this? what is this testing?

Comment (MegaRedHand) [reply .github/workflows/ci.yml:143]: It stops automagically. `mix run` starts and then stops the application (unless `--no-halt` is specified)

Comment (mpaulucci) [reply .github/workflows/ci.yml:143]: Maybe a comment or something would be nice, this is not clear from the command itself.

**Review by mpaulucci — APPROVED** [review]

**Review by mpaulucci — APPROVED** [review]
🚀

**Pull request merged**